### ISSUE TITLE
FPRTruncatedURLString   NSRangeException - *** -[__NSCFString substringToIndex:]: Inde out of bounds

### ISSUE THREAD

**Issue by evelyne96** [firebase/firebase-ios-sdk#13282]
### Description

We have recently started using Firebase performance and started seeing the following index out of bounds crash from the network trace library.
I was not able to reproduce this issue and did not find any older issues mentioning this. Was there any updates to fix this crash since version 10.24.0 ? 

```
          Fatal Exception: NSRangeException
0   CoreFoundation                 0x83f20 __exceptionPreprocess
1   libobjc.A.dylib                0x16018 objc_exception_throw
2   Foundation                     0x1150ec +[NSObject(NSKeyValueObservingPrivate) _shouldAddObservationForwardersForKey:]
3   PingerPhone                    0x71bae8 FPRTruncatedURLString + 119 (FPRDataUtils.m:119)
4   PingerPhone                    0x7222d0 +[FPRNetworkTrace stringByTrimmingURLString:] + 329 (FPRNetworkTrace.m:329)
5   PingerPhone                    0x720dec -[FPRNetworkTrace initWithURLRequest:] + 95 (FPRNetworkTrace.m:95)
6   PingerPhone                    0x728248 __InstrumentDataTaskWithRequestCompletionHandler_block_invoke + 316 (FPRNSURLSessionInstrument.m:316)
7   PingerPhone                    0x7281d4 __InstrumentDataTaskWithRequestCompletionHandler_block_invoke + 312 (FPRNSURLSessionInstrument.m:312)
8   OMSDK_Pinger                   0x15070 (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
9   OMSDK_Pinger                   0xd4cc (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
10  OMSDK_Pinger                   0xcedc (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
11  CoreFoundation                 0x20814 __invoking___
12  CoreFoundation                 0x1f860 -[NSInvocation invoke]
13  JavaScriptCore                 0x4a85a8 JSC::ObjCCallbackFunctionImpl::call(JSContext*, OpaqueJSValue*, unsigned long, OpaqueJSValue const* const*, OpaqueJSValue const**)
14  JavaScriptCore                 0x4a8094 JSC::objCCallbackFunctionCallAsFunction(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSValue*, unsigned long, OpaqueJSValue const* const*, OpaqueJSValue const**)
15  JavaScriptCore                 0x4a661c JSC::callObjCCallbackFunction(JSC::JSGlobalObject*, JSC::CallFrame*)
16  JavaScriptCore                 0x15772b8 pas_reallocation_did_fail
17  JavaScriptCore                 0xc6e408 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
18  JavaScriptCore                 0xf0cbe8 JSC::boundFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
19  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
20  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
21  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
22  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
23  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
24  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
25  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
26  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
27  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
28  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
29  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
30  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
31  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
32  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
33  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
34  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
35  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
36  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
37  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
38  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
39  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
40  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
41  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
42  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
43  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
44  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
45  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
46  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
47  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
48  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
49  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
50  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
51  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
52  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
53  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
54  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
55  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
56  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
57  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
58  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
59  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
60  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
61  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
62  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
63  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
64  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
65  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
66  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
67  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
68  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
69  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
70  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
71  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
72  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
73  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
74  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
75  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
76  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
77  JavaScriptCore                 0xc656c8 JSC::Interpreter::executeEval(JSC::EvalExecutable*, JSC::JSValue, JSC::JSScope*)
78  JavaScriptCore                 0xc63438 JSC::eval(JSC::CallFrame*, JSC::JSValue, JSC::JSScope*, JSC::ECMAMode)
79  JavaScriptCore                 0xdd0860 JSC::LLInt::commonCallDirectEval(JSC::CallFrame*, JSC::BaseInstruction<JSC::JSOpcodeTraits> const*, JSC::MacroAssemblerCodeRef<(WTF::PtrTag)1427>)
80  JavaScriptCore                 0xdd05d8 llint_slow_path_call_direct_eval
81  JavaScriptCore                 0x15a336c pas_reallocation_did_fail
82  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
83  JavaScriptCore                 0x15a19c4 pas_reallocation_did_fail
84  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
85  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
86  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
87  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
88  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
89  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
90  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
91  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
92  JavaScriptCore                 0xf0c4f8 JSC::boundThisNoArgsFunctionCall(JSC::JSGlobalObject*, JSC::CallFrame*)
93  JavaScriptCore                 0x15a39b0 pas_reallocation_did_fail
94  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
95  JavaScriptCore                 0x15a15a8 pas_reallocation_did_fail
96  JavaScriptCore                 0x15a03c0 pas_reallocation_did_fail
97  JavaScriptCore                 0x15771d4 pas_reallocation_did_fail
98  JavaScriptCore                 0xc6e504 JSC::Interpreter::executeCall(JSC::JSObject*, JSC::CallData const&, JSC::JSValue, JSC::ArgList const&)
99  JavaScriptCore                 0x4ca64c JSObjectCallAsFunction
100 JavaScriptCore                 0x46c5a0 -[JSValue callWithArguments:]
101 OMSDK_Pinger                   0xdfdc (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
102 OMSDK_Pinger                   0xd344 (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
103 OMSDK_Pinger                   0xd214 (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
104 OMSDK_Pinger                   0x15244 (Missing UUID 0686ed57c09c37caa3bd3c9b3aefc1f5)
105 libdispatch.dylib              0x213c _dispatch_call_block_and_release
106 libdispatch.dylib              0x3dd4 _dispatch_client_callout
107 libdispatch.dylib              0xb400 _dispatch_lane_serial_drain
108 libdispatch.dylib              0xbf30 _dispatch_lane_invoke
109 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh
110 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread
111 libsystem_pthread.dylib        0x4934 _pthread_wqthread
112 libsystem_pthread.dylib        0x10cc start_wqthread
        
```

### Firebase SDK Version

10.24.0

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

Performance

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by casperriboe** [comment]
I believe this was fixed in the latest release, 10.29.0. See this issue https://github.com/firebase/firebase-ios-sdk/issues/13199

**Comment by rizafran** [comment]
Hi @evelyne96, as mentioned by @casperriboe, this issues was fixed in the latest release. With this, please try updating your SDK to 10.29.0. Thanks.
